### DISCUSSION TITLE
Works on windows w/o WSL?

### DISCUSSION THREAD

**Discussion by charris-msft** [saulpw/visidata#1438] [General]
Hey folks, 
I've been an admirer of Visidata for some time but do all my work on Windows (and sometimes WSL). I could have been using Visidata, but didn't because of the extra WSL hoop (shame on me!)

After recently using some data engineering tools with pipx, I decided to try it with Visidata... 
`pipx install visidata`
and so far, it's worked beautifully!

Yes, I have WSL installed, but from a PowerShell prompt I can just type `visidata .\myawesome.csv`.

unfortunately, on my machine `vd` doesn't work. Windows just asks how do I want to open the file and looking at it I see a bit of python code:
```
#!C:\Users\charris\.local\pipx\venvs\visidata\Scripts\python.exe

import visidata.main

if __name__ == '__main__':
    visidata.main.vd_cli()
```

But who cares, this is awesome that it seems to just work! (and I could always create an alias for visidata.exe).

Has anyone tried this (rather than opening directly via WSL)? Are there any known issues?

**Comment by ajkerrigan** [comment]
Hiya @charris-msft , glad the pipx install worked outside WSL! Regarding the `vd` vs. `visidata` executable, I know it has come up in previous issues (see also #1406 and #1268) but I don't know what if anything could/should change to avoid that confusion on Windows.

As for running inside vs. outside WSL, my main suggestion would be to consider falling back to WSL if you hit any strange behavior. Pretty vague I know! But since much of VisiData's development and usage is Linux or Mac-focused, you might find some lesser-used features or plugins that make accidental platform assumptions.

**Comment by saulpw** [comment]
Hey @charris-msft, thanks for the post.  There are a few Windows users and my impression is that it works reasonably well.  There are some missing elements (like the underline attribute, used in both the column headers and to divide top/bottom split-screen), and some features that won't work without configuration.  I believe @aborruso and his crew use Windows, maybe they can chime in.

I'd love to have a proper Windows executable, ideally a plain downloadable .exe but it seems like that's not a reasonable thing to do these days on modern Windows. I did a basic pass in 2021 (which is in [visidata/vdplus](https://github.com/visidata/vdplus/tree/develop/windows)) to get a first version in the Microsoft App Store, and I ultimately gave up when it got caught in review--maybe this will become easier with [Microsoft's recent decision on commercial open source apps](https://techcrunch.com/2022/07/19/microsoft-u-turns-on-policy-that-wouldve-banned-commercial-open-source-apps/).  If you or anyone who uses Windows is interested in taking another pass at this, hit me up!

**Reply by aborruso** [reply]
Hi all, I have never used Windows.

I use VisiData in WSL for Linux and it works really great.

**Reply by nvallinoto** [reply]
dear all visidata lovers,
I just started a new edition of a open data course for a high superior school in genova and I have the following situation. 
I need to **install Visidata** in a **lab with 20 desktop pc having and 1 or more administrators and many simple (with no admin roles) users.**
This morning I tried to install Visidata with WSL (using an admin user) and until I use Visidata with this user the program works well.
The problem is that if I log again to the pc with a different and normal user Visidata is not working (and WSL too). As I understood from this discussion is not possible (https://github.com/microsoft/WSL/issues/5151) and the only solution is to install WSL for each user in that pc. So I need to install n(PC) multiply m(users) times WSL where n is the number of students and m is number of PC in the lab.. maybe you have better experience that simplify the situation ! Thanks for any help.

**Reply by nvallinoto** [reply]
another possible solution perhaps could be to install Visidata on Windows (without WSL) using an admin user the "pip3 install visidata" (after having installed python 3.11 version because with 3.12 is not working for Visidata..) with **cmd** or **powershell** with admininistrator option.
After that one can add to the system path the directory where visidata is installed (I guess it is the following folder but there is no visidata.py as I supposed) that is 
**C:\Users\vallinoto\AppData\Local\Programs\Python\Python311\Lib\site-packages\visidata**
And then logging again with another normal (with no admin role) user and see if visidata is working or not.
Maybe.

**Reply by nvallinoto** [reply]
I found the solution to install visidata in a laboratory with many multiuser pcs with windows 10/11 onboard. 

Here the steps for those having the same needs as me:

- Log in your PC with an admin user.

INSTALL PYTHON:
- download python 3.11 (https://www.python.org/ftp/python/3.11.9/python-3.11.9-amd64.exe)
- click on the exe running as administrator
- select the option "Add python.exe to PATH" and click on "Customize installation"
- select "for all users" 
- click next
- select "Install python 3.11 for all users"
- Customize install location should change in C:\Program files\Python311
- click install

INSTALL VISIDATA
- launch **cmd** or **powershell** running as administrator
- pip3 install visidata
- duplicate **visidata.exe** (from folder C:\Program files\Python311\Scripts) and rinominate in **vd.exe**

LAUNCH VISIDATA
- launch **cmd** or **powershell**
- write command line **vd** or **visidata** to open visidata

LAUNCH VISIDATA WITH ANOTHER USER 
- exit and log in with a normal user (with no admin role)
- launch **cmd** or **powershell**
- write command line **vd** or **visidata** to open visidata

CONFIGURATION FILE
- each user (let's say _myuser_) should have its own configuration file .visidatarc in personal home directory **c:\users\myuser**

THAT'S ALL!

**Reply by anjakefala** [reply]
Thanks so much for sharing your findings @charris-msft in step-by-step detail! I'm glad you found a reasonable process, and I'm really excited to hear about how your workshop goes.

What's `digit`?

**Reply by nvallinoto** [reply]
> Thanks so much for sharing your findings @charris-msft in step-by-step detail! I'm glad you found a reasonable process, and I'm really excited to hear about how your workshop goes.
> 
> What's `digit`?

"digit" means "write command line" VD or VISIDATA to launch the program

this morning I just started a new edition (the 5th) of a course on open data with two classes of technical institute Calvino of Genova.
Here you can find information about the first 4 editions took place in different schools of my town.
https://www.scuoladigitaleliguria.it/community/pcto/pcto-open-data.html

After two lessons on theory of open data we (me and other colleagues) submit practice lessons showingthe students two free software to manage and visualize data: VisiData and QGis. And last but not least the students divide themselves in little groups and make exercises using open data with VisiData and QGis.

**Comment by saulpw** [comment]
Also how do you feel about making this a general Windows discussion thread?  I'd love a place to point people with issues like #1431 for a broader discussion.

**Reply by charris-msft** [reply]
I love the idea!

**Reply by nvallinoto** [reply]
yes I think it's a good idea @saulpw

**Comment by danihinder** [comment]
Have you ever tried to replay a [command log](https://jsvine.github.io/intro-to-visidata/advanced/the-command-log/)? Doesn't work for me and I'm not sure if this is because of I'm on windows or something else. It just does nothing when I invoke the --play option. I should/will try this in WSL, maybe it's just me :)

**Reply by danihinder** [reply]
doesn't work there as well, it's just me....

**Reply by saulpw** [reply]
Hi @danihinder, can you share the version of VisiData, the cmdlog, possibly the dataset, and the exact command line used to run it?

**Reply by danihinder** [reply]
Hi Saul, while preparing a sized down test example, I discovered something: It works with a .tsv dataset as a source, I'm pretty happy now! I just discovered Visidata and it is so cool! I'm able to reproduce the error with a .xlsx file which I'll try to attach.
The command line is `visidata --play test_cmdlog.vdj`

[test_source.xlsx](https://github.com/saulpw/visidata/files/9914054/test_source.xlsx)

[test_cmdlog.vdj.txt](https://github.com/saulpw/visidata/files/9914061/test_cmdlog.vdj.txt)

**Reply by danihinder** [reply]
And the version of VisiData is v2.10.2

**Reply by danihinder** [reply]
Hi @saulpw, I just tried it with a .xls dataset and it doesn't work with this either. I'm about to try it with an .ods, but can't figure out what to install to get rid of "ModuleNotFoundError: No module named 'odf'" during opening a .ods File. Can't find a hint in the [docs](https://www.visidata.org/docs/formats/) and not [here](https://jsvine.github.io/intro-to-visidata/basics/opening-files/). I'm to new in town to find a way.

**Reply by anjakefala** [reply]
Hi @danihinder! How you would update modules depends on how VisiData was installed in the first place. Do you remember how you installed VisiData?

**Reply by danihinder** [reply]
I installed it with `pip install visidata`, to get excel sheets working, I installed `pip install xlrd openpyxl`

**Reply by anjakefala** [reply]
Yup! And to get `.ods` working, you install [odfpy](https://github.com/saulpw/visidata/blob/924563a0b7eeede80834dc02d8c1f237fe1949c1/requirements.txt#L30).

You are right that it would be nice if we could add documentation on installing optional modules to the install page.

**Reply by danihinder** [reply]
I should have mentioned, I already tried `pip install odfpy` to no avail. But I'm actually pretty new to python, so assume the worst with me :)

**Reply by danihinder** [reply]
Nevermind, I just installed `pip install odfpy` in a WSL SuSE and it's working fine there. So it's something self made in my Windows installation which I'll figure out then for myself. No worries, the documentation is top notch, the "odfpy" part was missing, that's all.
I just made a test in WSL with an .odf file regarding the cmdlog problem, it doesn't work there. It's exactly the same behaviour like with the .xls file.

**Reply by danihinder** [reply]
I just found out what I did wrong: If I had opened a file in an already running visidata instance, the subsequent playback of the cmdlog did not work, because the file name in the cmdlog is missing. If I open the same file at the command line with `visidata ./some_file.xlsx`, the subsequent playback of the cmdlog works. Stupid beginner's mistake, I could have recognized it earlier in the cmdlog, it's there in plain text. Now everything really works and I am more than happy! Sorry for the confusion with .xlsx .ods etc... I assumed a bug, but I'm glad, it was really just me :)

**Reply by jabellcu** [reply]
I had the same issue and this post has helped me realize what the problem was. Thanks!
